code/modules/clothing/clothing.dm

```diff
@@ -79,15 +90,107 @@
+  *
+  * Arguments:
+  * * def_zone: The bodypart zone in question
+  * * damage_amount: Incoming damage
+  * * damage_type: BRUTE or BURN
+  * * armour_penetration: If the attack had armour_penetration
+  */
+/obj/item/clothing/proc/take_damage_zone(def_zone, damage_amount, damage_type, armour_penetration)
+	if(!def_zone || !limb_integrity || (initial(body_parts_covered) in GLOB.bitflags)) // the second check sees if we only cover one bodypart anyway and don't need to bother with this
+		return
+	var/list/covered_limbs = body_parts_covered2organ_names(body_parts_covered) // what do we actually cover?
+	if(!(def_zone in covered_limbs))
+		return
+
+	var/damage_dealt = take_damage(damage_amount * 0.1, damage_type, armour_penetration, FALSE) * 10 // only deal 10% of the damage to the general integrity damage, then multiply it by 10 so we know how much to deal to limb
+	LAZYINITLIST(damage_by_parts)
+	damage_by_parts[def_zone] += damage_dealt
+	if(damage_by_parts[def_zone] > limb_integrity)
+		disable_zone(def_zone, damage_type)
+
+/**
+  * disable_zone() is used to disable a given bodypart's protection on our clothing item, mainly from [/obj/item/clothing/proc/take_damage_zone()]
+  *
+  * This proc disables all protection on the specified bodypart for this piece of clothing: it'll be as if it doesn't cover it at all anymore (because it won't!)
+  * If every possible bodypart has been disabled on the clothing, we put it out of commission entirely and mark it as shredded, whereby it will have to be repaired in
+  * order to equip it again. Also note we only consider it damaged if there's more than one bodypart disabled.
+  *
+  * Arguments:
+  * * def_zone: The bodypart zone we're disabling
+  * * damage_type: Only really relevant for the verb for describing the breaking, and maybe obj_destruction()
+  */
+/obj/item/clothing/proc/disable_zone(def_zone, damage_type)
+	var/list/covered_limbs = body_parts_covered2organ_names(body_parts_covered)
+	if(!(def_zone in covered_limbs))
+		return
+
+	var/zone_name = parse_zone(def_zone)
+	var/break_verb = ((damage_type == BRUTE) ? "torn" : "burned")
+
+	if(iscarbon(loc))
+		var/mob/living/carbon/C = loc
+		C.visible_message("<span class='danger'>The [zone_name] on [C]'s [src.name] is [break_verb] away!</span>", "<span class='userdanger'>The [zone_name] on your [src.name] is [break_verb] away!</span>", vision_distance = COMBAT_MESSAGE_RANGE)
+		RegisterSignal(C, COMSIG_MOVABLE_MOVED, .proc/bristle)
+
+	zones_disabled++
+	for(var/i in zone2body_parts_covered(def_zone))
+		body_parts_covered &= ~i
+
+	if(body_parts_covered == NONE) // if there are no more parts to break then the whole thing is kaput
+		obj_destruction((damage_type == BRUTE ? "melee" : "laser")) // melee/laser is good enough since this only procs from direct attacks anyway and not from fire/bombs
+		return
+
+	switch(zones_disabled)
+		if(1)
+			damaged_clothes = CLOTHING_PRISTINE // i mean it's not pristine but it's not super damaged

```

Comment:
this means that you can't repair it

... unless you INTENTIONALLY DAMAGE YOUR OWN CLOTHES to damage a second region, allowing you to repair the clothing

Reply:
removed